core/mondoo-aws-security.mql.yaml

```diff
@@ -1165,4 +1076,5 @@ queries:
       remediation: "__Terraform__\n\n```hcl\nTODO\n```\n\n__AWS Console__\n\nTo enable encryption for CloudTrail logs\n\n  1. Open the CloudTrail console at https://console.aws.amazon.com/cloudtrail/.\n  2. Choose Trails.\n  3. Choose the trail to update.\n  4. Under General details, choose Edit.\n  5. For Log file SSE-KMS encryption, select Enabled.\n  6. For Create a new KMS key, do one of the following:\n  \n    * To create a key, choose New. Then in AWS KMS alias, enter an alias for the key. The key is created in the same Region as the S3 bucket.\n    * To use an existing key, choose Existing, then from AWS KMS alias, choose the key.\n    * The AWS KMS key and S3 bucket must be in the same Region.\n\n  7. Choose Save.\n\n__AWS CLI__\n\n```bash\nTODO\n```\n"
   - uid: mondoo-aws-security-secgroup-restricted-ssh
     title: Checks that all incoming SSH traffic for the security groups is restricted
-    mql: aws.ec2.securitygroup { ipPermissions.all(ipRanges.length == 0 && fromPort == 0 && toPort == 0)}
+    impact: 80
+    mql: aws.ec2.securityGroups.where( ipPermissions { toPort == 22 }).all( ipPermissions.all( ipRanges.none("0.0.0.0/0") ))
```

Comment:
block calls should not be used in where conditions

Reply:
@chris-rock the current version of the query now works much better. Can you help me to understand why the of blocks in a `where` function is problematic? 🤔 

## Current version
<img width="779" alt="image" src="https://github.com/mondoohq/cnspec-policies/assets/49754039/7dd93661-6602-46be-b775-5a65c17e2ffa">

## Proposed change
![image](https://github.com/mondoohq/cnspec-policies/assets/49754039/c6f7236b-44d1-4e35-81c8-9115cba64a7c)

Reply:
conditions in '.where(condition)' should be a clear Boolean expression. Using blocks makes it more difficult for mql reader to understand if the condition matches or not

Reply:
@chris-rock I chatted with @arlimus on this one and he shared that blocks `{}` are not technically boolean. The updated query:

```typescript
aws.ec2.securityGroups.where( ipPermissions.contains( toPort == 22 )).all( ipPermissions.all( ipRanges.none("0.0.0.0/0") ))
```

Reply:
i just went through and found these changes. those single platform queries were for the fine grained assets, that's why they had the fine grained asset platform filter on them. i guess it'll make more sense once we have variants 🤷